src/LibraryManager.Vsix/Json/TextviewCreationListener.cs

```diff
@@ -81,17 +81,21 @@ public void VsTextViewCreated(IVsTextView textViewAdapter)
                 foreach (ILibraryInstallationState state in manifest.Libraries.Where(l => l.IsValid(out IEnumerable<IError> errors)))
                 {
                     IProvider provider = _dependencies.GetProvider(state.ProviderId);
-                    ILibraryInstallationResult updatedStateResult = provider.UpdateStateAsync(state, CancellationToken.None).Result;
 
-                    if (updatedStateResult.Success)
+                    if (provider != null)
```

Comment:
Should we give a "Bad provider id" error or something like that? If the user accidentally mistypes the id, it would be good for them to know what's wrong.

Reply:
Yes, we do give a proper error message. This check is only during file deletion before restore. At restore time, we give the proper validation errors.